### PR TITLE
[Filebeat] Add Malware Bazaar to Threat Intel Module

### LINKED ISSUE #24569: [Filebeat] Malware Bazaar Dataset for Threat Intel Module
## Modules

**Describe the enhancement:**
Currently, the threat intel module for Filebeat did not have the data provided by Malware Bazaar.

Malware Bazaar provides rich file metadata about malware that can assist cyber intelligence analysts, threat hunters, and incident responders during incident response and ongoing security operations.

**Describe a specific use case for the enhancement or feature:**
Threat hunting, security operations, and intelligence analysis.

## Filebeat module

* [x] Test log files exist for the grok patterns
* [x] Generated output for at least 1 log file exists

### PR DESCRIPTION
**Pull request by peasead**
## What does this PR do?
This PR adds the Malware Bazaar threat feed to the threat intel module of Filebeat.

## Why is it important?
Malware Bazaar provides rich file metadata about malware that can assist cyber intelligence analysts, threat hunters, and incident responders during incident response and ongoing security operations.

Currently, the threat intel module for Filebeat did not have the data provided by Malware Bazaar.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [x] Validated that `./filebeat setup` imports the saved search, visualization, and dashboard
- [x]  Validated that data provided by the Threat Intel Filebeat module populates the dashboards

## How to test this PR locally

This can be tested by:
```
cd beats/x-pack/filebeat
mage update && mage build
./filebeat modules enable threatintel
# if using Elastic Cloud, update filebeat.yml to cloud.id and cloud.auth
./filebeat setup -E setup.dashboards.directory=./build/kibana
./filebeat
```
Go to KIbana -> dashboards -> apply the `threat intel` tag

## Related issues

Resolves #24569 

## Use cases

Threat hunting, security operations, and intelligence analysis.

## Screenshots
filebeat-threatintel-malware-bazaar

### Original data from source
```
  {
            "sha256_hash": "ff4ad9465d1312e24dff76c47ddb519bc46f501239e5813d4d70ac6bd7c70638",
            "sha3_384_hash": "40ce04e3e695d10b42d9ca1026476aa6076923a911652c6f263f866c8ce86e9bf2b03e1626f3ce20f6d4f1649ed3f54a",
            "sha1_hash": "eb519d0338c102103d0e8467a5dba55c99a40233",
            "md5_hash": "4f0f5e2c6bc8d65d14ff201d799824ae",
            "first_seen": "2021-03-16 16:07:32",
            "last_seen": null,
            "file_name": "tarifvertrag_chemie_bayern_entgelttabelle.js",
            "file_size": 2939,
            "file_type_mime": "text\/plain",
            "file_type": "js",
            "reporter": "dor0n1",
            "origin_country": "DE",
            "anonymous": 0,
            "signature": null,
            "imphash": null,
            "tlsh": "6E51A6E5B149E15101073734255FE86CB9B3A949D40ED420D749D7DF28B603D4E27A9E",
            "telfhash": null,
            "ssdeep": "48:M27hDQrbyxd8CROAdiQD0GMFJ0QXtAkiQKXJXWcaiMKjTlLJ\/9zhjOquc9+ac03V:5Vwbg8BxHpX4XMAF96qfmaMJ0ncTwy1w",
            "tags": [
                "gootloader"
            ],
            "code_sign": [],
            "intelligence": {
                "clamav": null,
                "downloads": "15",
                "uploads": "1",
                "mail": null
            }
        },
```

### Data from module

```
{
  "_index": "filebeat-8.0.0-2021.03.16-000001",
  "_type": "_doc",
  "_id": "38de8b728560e8705f707d4607706d73f04b4d50f925608ffed7b873e66e9b67",
  "_version": 1,
  "_score": null,
  "_source": {
    "agent": {
      "name": "c02zc1eklvdr",
      "id": "4aa831e8-28aa-4037-ab92-02cdcdf0e6ae",
      "ephemeral_id": "c6efb3b7-007f-42ad-b2bf-c34fb577a9a7",
      "type": "filebeat",
      "version": "8.0.0"
    },
    "fileset": {
      "name": "malwarebazaar"
    },
    "threatintel": {
      "malwarebazaar": {
        "intelligence": {
          "mail": {
            "CH": "low",
            "Generic": "high"
          },
          "downloads": 22,
          "uploads": 1
        },
        "tags": [
          "doc"
        ],
        "anonymous": 0,
        "code_sign": []
      },
      "indicator": {
        "geo": {
          "country_iso_code": "FR"
        },
        "first_seen": "2021-03-16T16:05:36.000Z",
        "file": {
          "extension": "doc",
          "size": 167472,
          "mime_type": "text/rtf",
          "pe": {},
          "name": "Purchase order.doc",
          "hash": {
            "sha1": "1471a4b801b065b315f130a5d512dc7597c71657",
            "sha384": "c70be406ff428593018f85cc8363635c259506197972371e24f7161011a8a46dca83ac0268ca80d8a92d6ccf93fcc9a1",
            "sha256": "623119afef4f91cf052823dd4d6b1ec9f04dc7fbe76a313035e0e4aa00888454",
            "tlsh": "30F36DA8E991CDD4CBCFC9D94A1E3A952033FA79C6C36C964438F3F50B9267F4A16850",
            "ssdeep": "3072:Cv6q8aN9Zts1unvhfKwGt7D2l8hnoo+ixxpbeBJTJzEmFhThl4O/b+8OGo3VaESF:CvkaN9Zts6vVGt/2l8nxpKJTJznhdDTP",
            "md5": "a24571eb4baea7a5b67480d524668617"
          }
        },
        "provider": "cocaman",
        "type": "file"
      }
    },
    "tags": [
      "threatintel-malwarebazaar",
      "forwarded"
    ],
    "input": {
      "type": "httpjson"
    },
    "@timestamp": "2021-03-16T16:37:32.980Z",
    "ecs": {
      "version": "1.6.0"
    },
    "related": {
      "hash": [
        "a24571eb4baea7a5b67480d524668617",
        "623119afef4f91cf052823dd4d6b1ec9f04dc7fbe76a313035e0e4aa00888454",
        "3072:Cv6q8aN9Zts1unvhfKwGt7D2l8hnoo+ixxpbeBJTJzEmFhThl4O/b+8OGo3VaESF:CvkaN9Zts6vVGt/2l8nxpKJTJznhdDTP"
      ]
    },
    "service": {
      "type": "threatintel"
    },
    "event": {
      "ingested": "2021-03-16T16:37:34.071668028Z",
      "created": "2021-03-16T16:37:32.980Z",
      "kind": "enrichment",
      "module": "threatintel",
      "category": "threat",
      "type": "indicator",
      "dataset": "threatintel.malwarebazaar"
    }
  },
  "fields": {
    "event.ingested": [
      "2021-03-16T16:37:34.071Z"
    ],
    "@timestamp": [
      "2021-03-16T16:37:32.980Z"
    ],
    "event.created": [
      "2021-03-16T16:37:32.980Z"
    ]
  },
  "sort": [
    1615912652980
  ]
},
```

### CODE REVIEW

**Comment by P1llus** [comment]
jenkins run tests please

**Comment by P1llus** [comment]
The timeout error from the build can either be from the CI, or a typo somewhere in your parser. I restarted the job for now @peasead , but if it fails again then you might need to try to rerun some of your tests locally.

**Comment by P1llus** [comment]
The format of the test files is a bit wrong, working with @peasead to resolve that.

**Comment by P1llus** [comment]
You would need to add the fileset to the timestamp ignore list, That's found here: https://github.com/elastic/beats/blob/master/filebeat/tests/system/test_modules.py#L276

File: x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    field: related.hash
+    value: '{{ threatintel.indicator.file.hash.ssdeep }}'
+    if: ctx?.threatintel?.indicator?.file?.hash?.ssdeep != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.pe.imphash }}'
+    if: ctx?.threatintel?.indicator?.file?.pe?.imphash != null
+- append:
+    field: related.hash
+    value: '{{ threatintel.indicator.file.pe.tlsh }}'
```
Comment (andrewstucki) [inline x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml:146]: I'll see if maybe we can put this in as an official ECS field, but just wondering, why `pe.tlsh`? `tlsh` is a generic fuzzy hash, so it should pretty much be wherever `ssdeep` is.

Comment (andrewstucki) [reply x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml:146]: also, it doesn't look like it's in the pipeline, and I'm sure we could look into adding it in to the `elf` RFC, but what about `telfhash`? Thinking we could map it to `threatintel.indicator.file.elf.telfhash` and probably dump into `related.hash` as well.

Comment (peasead) [reply x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml:146]: > I'll see if maybe we can put this in as an official ECS field, but just wondering, why `pe.tlsh`? `tlsh` is a generic fuzzy hash, so it should pretty much be wherever `ssdeep` is (i.e. `hash.tlsh`).

I see now. I copied the existing ingest pipeline from `abusemalware` and that appears to be where the issue is. `abusemalware` and `malwarebazaar` need to use `*.hash.tlsh` vs. `*.pe.tlsh`.
```
- append:
    field: related.hash
    value: '{{ threatintel.indicator.file.hash.tlsh }}'
    if: ctx?.threatintel?.indicator?.file?.hash?.tlsh != null
```

I'll make this correction and open an Issue for `abusemalware`.

Comment (peasead) [reply x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml]: > also, it doesn't look like it's in the pipeline or filled in in your source documents (just looks like it's `null` everywhere), and I'm sure we could look into adding it in to the `elf` RFC, but what about `telfhash`? Thinking we could map it to `threatintel.indicator.file.elf.telfhash` and probably dump into `related.hash` as well.

I added `*.elf.telfhash`, good addition.

Comment (peasead) [reply x-pack/filebeat/module/threatintel/malwarebazaar/ingest/pipeline.yml]: `tlsh` bug for Abuse Malware
https://github.com/elastic/beats/issues/25151

**Review by P1llus — APPROVED** [review]
Approved for now after the changes from Andrew K is in and the build passes. Any small updates we might want to do would need to be added after FF.

**Pull request merged**